### PR TITLE
[Html2] Fix circular references

### PR DESCRIPTION
**Pull request by wy-z**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/swagger-api/swagger-codegen/blob/master/CONTRIBUTING.md).
- [x] Ran the shell/batch script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates)
- [x] Filed the PR against the correct branch: master for non-breaking changes and `2.3.0` branch for breaking (non-backward compatible) changes.

### Description of the PR

- Replace `jsonref` with [`BigstickCarpet/json-schema-ref-parser`](https://github.com/BigstickCarpet/json-schema-ref-parser)
- Replace `JSON.stringify` with [`stringify_safe`](https://github.com/isaacs/json-stringify-safe)

### CODE REVIEW

**Comment by wy-z** [comment]
Spec for test.
[swagger.json.zip](https://github.com/swagger-api/swagger-codegen/files/892065/swagger.json.zip)

**Comment by wing328** [comment]
@wy-z thanks for the PR.

@jhitchcock, please take a look when you've time.

File: modules/swagger-codegen/src/main/resources/htmlDocs2/paramB.mustache [hunk 1]
```diff
@@ -3,33 +3,24 @@
...
-
-												});
-
-
-
-												});
-												</script>
-												<div id="d2e199_{{nickname}}_{{paramName}}"></div>
+$(document).ready(function() {
+	var schemaWrapper = {{{jsonSchema}}};
```
Comment (wing328) [inline modules/swagger-codegen/src/main/resources/htmlDocs2/paramB.mustache]: @wy-z Shall we use 2-space instead of tab for code indentation?

Comment (wy-z) [inline modules/swagger-codegen/src/main/resources/htmlDocs2/paramB.mustache]: Done.

**Comment by wing328** [comment]
Thanks for the PR. If your company is using Swagger Codegen, please consider adding your company name to [this list](https://github.com/swagger-api/swagger-codegen#companiesprojects-using-swagger-codegen)

**Comment by wy-z** [comment]
It's our pleasure to do this~
@wing328  #5418

**Pull request merged**